auth/auth/auth.py

```diff
@@ -737,11 +743,49 @@ async def rest_logout(request: web.Request, userdata: UserData) -> web.Response:
     return web.Response(status=200)
 
 
-async def get_userinfo(request: web.Request, session_id: str) -> UserData:
+async def get_userinfo(request: web.Request, auth_token: str) -> UserData:
+    flow_client: Flow = request.app['flow_client']
+    client_session = request.app['client_session']
+
+    userdata = await get_userinfo_from_hail_session_id(request, auth_token)
+    if userdata:
+        return userdata
+
+    hailctl_oauth_client = request.app['hailctl_client_config']
+    uid = await flow_client.get_identity_uid_from_access_token(
+        client_session, auth_token, oauth2_client=hailctl_oauth_client
+    )
+    if uid:
+        return await get_userinfo_from_login_id_or_hail_identity_id(request, uid)
+
+    raise web.HTTPUnauthorized()
+
+
+async def get_userinfo_from_login_id_or_hail_identity_id(request: web.Request, login_id: str) -> UserData:
```

Comment:
Can you change the variable name to not be login_id since it could be that or the hail identity?